ipaclient/plugins/vault.py

```diff
@@ -118,6 +118,8 @@ def encrypt(data, symmetric_key=None, public_key=None):
                 label=None
             )
         )
+    else:
+        return None
```

Comment:
`encrypt()` cannot return None because we call `base64.b64encode()` on the result unconditionally.

```
>>> import base64
>>> base64.b64encode(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.6/base64.py", line 58, in b64encode
    encoded = binascii.b2a_base64(s, newline=False)
TypeError: a bytes-like object is required, not 'NoneType'
>>> 
```

I think for both `encrypt()` and `decrypt()` we can return `b''` or similar.

Reply:
No, returning b'' is wrong and will hide bugs.

We have returned None in the past. This change just makes it obvious and explicit.

Reply:
then I'd suggest to raise an error here. Since both `encode` and `decode` cannot function without symmetric or public/private keys, raising an exception is better than failing in base64.